pkg/pgmodel/sql_reader.go

```diff
@@ -259,8 +158,10 @@ type timescaleRow struct {
 	err      error
 }
 
+// getResultRows fetches the result row datasets from the database using the
+// supplied query parameters.
 func (q *pgxQuerier) getResultRows(startTimestamp int64, endTimestamp int64, hints *storage.SelectHints, path []parser.Node, matchers []*labels.Matcher) ([]timescaleRow, parser.Node, error) {
-
+	// Build a subquery per metric matcher.
 	metric, cases, values, err := buildSubQueries(matchers)
```

Comment:
Should we rename `buildSubQueries` to `buildMetricQuery` as subquery confuses me with the ones in our promql package?

Reply:
The problem is that its not a metric query, its a sub-query clause so calling it a metric query makes no sense.